components/player/Comments.tsx

```diff
@@ -32,25 +40,71 @@ const Name = styled.span`
   color: rgba(255, 255, 255, 0.9);
 `
 
+const Button = styled<'div', {type: COMMENT_MENU}>('div')`
+  width: 32px;
+  height: 32px;
+  border-radius: 50%;
+  background-image: url(${({ type }) => `/images/${COMMENT_MENU_ICON[type]}`});
+  background-repeat: no-repeat;
+  background-position: center;
+  background-color: ${({ type }) => COMMENT_MENU_COLOR[type]};
+
+  &:first-of-type {
+    margin-bottom: 10px;
+  }
+`
+
 export const Comments = ({ kind, contents, time }: Props) => {
+  const [{ x }, set] = useSpring(() => ({ x: 0 }))
+
+  const bind = useDrag(({ movement: [mx], last }) => {
+
+    // last: mouseUp
+    const newX = last ? getClosest([-100, 0], x.getValue()) : mx
+
+    set({ x: newX })
+  }, {
+    bounds: { left: -100, right: 100 },
+    rubberband: true, // 경계 벗어날때 탄성계수
+  })
+
   return (
     <Wrapper>
-      <ContentWrapper>
-        <ProfileImage />
-        <div>
-          <Name>{COMMENT_KIND[kind]}</Name>
-          {contents.map((content: string, index: number) => {
-            return (
-              <SpeechBubble
-                key={`${kind}-${index}`}
-                time={time}
-                kind={kind}
-                content={content}
-              />
-            )
-          })}
-        </div>
-      </ContentWrapper>
+      <animated.div {...bind()} style={
+        {
+          transform: x.interpolate(x => `translateX(${x}px)`),
+        }
+      } >
+        <ContentWrapper>
+          <ProfileImage />
+          <div>
+            <Name>{COMMENT_KIND[kind]}</Name>
+            {contents.map((content: string, index: number) => {
```

Comment:
```suggestion
            {contents.map((content, index) => {
```

we don't need to typing because typescript can infer `content` type and `index` type